### DISCUSSION TITLE
package-lock.json keeps stale resolutions of git/github dependencies

### DISCUSSION THREAD

**Discussion by pzoomin** [npm/feedback#869] [CLI] [answered]
Using a git or github url for a dependency, as documented at https://docs.npmjs.com/cli/v9/configuring-npm/package-json#github-urls, works the first time, but for any subsequent changes to that dependency it keeps a stale resolution.

Eg, in a projects's `package.json`, if we have a section like this, pretending there is a github project at `example/myDependency", that we pull into our project as so:

```
"dependencies": {
  "myDependency": "example/myDependency#testingBranch"
}
```

And we run `npm install`, then checking both the `node_modules` and `package-lock.json`, we see the correct code for that dependency is installed. But if we then make a change:

```
"dependencies": {
  "myDependency": "example/myDependency#anotherBranch"
}
```

And we run `npm install`, checking `node_modules` and `package-lock.json` reveals that we are *NOT* using the code from the `anotherBranch` branch, but still using the code from `testingBranch`.

This can be "solved" by deleting `package-lock.json` and the dependency from `node_modules`, but this seems like a bug in NPM's resolution strategy.

**Comment by MylesBorins** [comment, marked as answer]
This should be opened as a bug report over at https://github.com/npm/cli
